File: youlexuan-goods-interface/src/main/java/com/offcn/goods/service/GoodsService.java
package com.offcn.goods.service;

import com.offcn.pojo.TbGoods;
import com.offcn.vo.resp.GoodsVO;
import com.offcn.vo.resp.PageResult;

import java.util.List;

public interface GoodsService {

    /**
     * 添加商品组合实体类
     * @param goodsVO
     */
    public void add(GoodsVO goodsVO);

    /**
     * 分页查询商品列表
     * @param pageNum
     * @param pageSize
     * @param searchEntity
     * @return
     */
    public PageResult getGoodsList(int pageNum,int pageSize,TbGoods searchEntity,String sellerId);

    /**
     * 批量审核商品
     * @param ids
     * @param status
     */
    public void updateGoodsStatus(Long[] ids,String status);

    /**
     * 批量删除商品
     * @param ids
     */
    public void deleteGoods(Long[] ids);

    /**
     * 获取品牌名称显示在商品详情页面
     * @param id
     * @return
     */
    public String getBrandName(Long id);

    /**
     * 商家后台对商品进行上架和下架的操作
     * @param isId
     */
    public void updateMarket(Long id,String isId);
}

File: youlexuan-manage-web/src/main/webapp/js/service/goodsService.js
app.service('goodsService',function ($http) {

    //分页加载商品列表
    this.getGoodsList=function (url,entity) {
        return $http.post(url,entity);
    };
    //商品名称数组
    this.getItemCatList=function () {
        return $http.get("/goods/getItemCatList.do");
    };
    //修改商品状态
    this.updateGoodsStatus=function (ids,status) {
        return $http.post("/goods/updateGoodsStatus.do?ids="+ids+"&status="+status);
    };
    //批量删除
    this.deleteGoods=function (ids) {
        return $http.post("/goods/deleteGoods.do?ids="+ids);
    };
    //获取品牌名称
    this.getBrandName=function (id) {
        return $http.get("/goods/getBrandName.do?id="+id);
    }
});